PokemonRampUp/Model/Network.swift

```diff
@@ -0,0 +1,29 @@
+//
+//  Network.swift
+//  PokemonRampUp
+//
+//  Created by Thushen Mohanarajah on 2023-02-14.
+//
+
+import Foundation
+
+// TO-DO [POKEMON-0012] Create unit test for Network model
+class Network{
+    
+    static func loadJSONObject<T: Decodable>(stringURL: String, type: T.Type) async throws -> T{
+        guard let url = URL(string: stringURL) else {
+            throw NetworkError.invalidURL
+        }
+        
+        guard let (data, _) = try? await URLSession.shared.data(from: url) else{
+            throw NetworkError.requestError
+        }
+        
+        let decoder = JSONDecoder()
+        guard let objectT = try? decoder.decode(T.self, from: data) else {
+            throw NetworkError.dateParseError
+        }
+        
+        return objectT
+    }
+}
```

Comment:
- It's a good practice to abstract the interfaces from their implementations in protocols, that way you avoid your classes depend on concrete implementations and also facilitates the writing of unit tests in the classes that will need to interact with your `Network` component.
- Avoid using static functions unless strictly necessary.

Reply:
```suggestion
protocol Network {
    func loadJSONObject<T: Decodable>(stringURL: String, type: T.Type) async throws -> T
}

final class DefaultNetwork: Network {
    private let urlSession: URLSession
    
    init(urlSession: URLSession = .shared) {
        self.urlSession = urlSession
    }

     func loadJSONObject<T: Decodable>(stringURL: String, type: T.Type) async throws -> T {
         guard let url = URL(string: stringURL) else {
             throw NetworkError.invalidURL
         }

         guard let (data, _) = try? await urlSession.data(from: url) else{
             throw NetworkError.requestError
         }

         let decoder = JSONDecoder()
         guard let objectT = try? decoder.decode(T.self, from: data) else {
             throw NetworkError.dateParseError
         }

         return objectT
     }
 }
```

Reply:
The `Network` uses a URLSession to send requests. Its constructor takes a URLSession instance, and the default value is URLSession.shared. As a result, most of the time, you can generate new instances of your networking object without explicitly passing the URL session. If you decide to use a distinct URL session, such as in your unit tests, you can easily pass the session to the networking object's constructor, and it will use your custom URL session.

Reply:
That's a really good idea! I was actually wondering how I would go about testing this file because, I don't have direct access to URLSession instance. Now I can easily just create a mock URLSession and initialize it in the constructor. Thanks for the explanation!